modules/javafx.controls/src/main/java/javafx/scene/control/Control.java

```diff
@@ -222,7 +222,12 @@ private static Class<?> loadClass(final String className, final Object instance)
      * {@code Skin}. Every {@code Skin} maintains a back reference to the
      * {@code Control} via the {@link Skin#getSkinnable()} method.
      * <p>
+     * To ensure a one-to-one relationship between a {@code Control} and its {@code Skin},
+     * {@link Control#setSkin(Skin)} method will check the return value of {@link Skin#getSkinnable()}
+     * against this Control, and throw an {@code IllegalArgumentException} if it is not the same.
+     * <p>
      * A skin may be null.
+     *
      * @return the skin property for this control
```

Comment:
I recommend adding an `@throws` clause here. Something like:

```
     * @throws IllegalArgumentException if {@code skin != null && skin != getSkinnable()}
```

although the javadoc tool currently doesn't do anything with an `@throws` tag in a property (I'll file a follow-up bug for this).